### ISSUE TITLE
RFC [16.0] shopfloor* - Documentation Initiative - How Can We Get Involved?

### ISSUE THREAD

**Issue by luke-stdev001** [OCA/wms#926]
We are working on rolling out the shopfloor to replace the Odoo core barcode scanner for our warehouse and retail store operations. During this process we will be creating our own internal documentation for usage guidelines, configuration, etc.

Would it be useful for us to submit our documentation efforts to the OCA in a PR, and are there any guidelines, structure or format that should be followed?

We're unaware of any documentation for the shopfloor* modules and WMS related modules so we will create our own, we'd just like to contribute back if this is something the OCA is interested in.

**Comment by rousseldenis** [comment]
> We're unaware of any documentation for the shopfloor* modules and WMS related modules so we will create our own, we'd just like to contribute back if this is something the OCA is interested in.

Thanks for the initiative !

In fact, there is already some documentation there: https://github.com/OCA/wms/tree/16.0/shopfloor/docs
with images and UML diagrams.

If you come to OCA days, we can talk together to get further.

**Comment by luke-stdev001** [comment]
> > We're unaware of any documentation for the shopfloor* modules and WMS related modules so we will create our own, we'd just like to contribute back if this is something the OCA is interested in.
> 
> Thanks for the initiative !
> 
> In fact, there is already some documentation there: https://github.com/OCA/wms/tree/16.0/shopfloor/docs with images and UML diagrams.
> 
> If you come to OCA days, we can talk together to get further.

Thanks @rousseldenis ! I somehow overlooked that folder.

What I was planning on doing is putting together some functional documentation for user flows with the shopfloor modules. We have our own internal repo for documentation on sphinx which focuses on training material for our retail store teams, i'm just wondering if while we are doing this would the OCA be interested in any PR's for functional docs for using the barcode gun.

We also plan to add a few new features such as Inventory adjustments based on stock_cycle_count, something like shopfloor_stock_cycle_count, would this be something the OCA would be interested in receiving a PR for? We have a similar initiative for printing ZPL product price labels using shopfloor as our base with direct print through the modules in report-print-send and our own additions.

**Comment by rousseldenis** [comment]
> We also plan to add a few new features such as Inventory adjustments based on stock_cycle_count, something like shopfloor_stock_cycle_count, would this be something the OCA would be interested in receiving a PR for?

Of course. We had not migrated I think the normal inventory part to v16 and that should be great to begin by that part (as I suppose cycle_count is an exension of it)

**Comment by rousseldenis** [comment]
> i'm just wondering if while we are doing this would the OCA be interested in any PR's for functional docs for using the barcode gun.

There is at OCA a new "functional" group composed essentially by functional people. Their objective is to enhance OCA procedures, documentation to attract more people using and contributing to OCA. Maybe that could fit their needs.

**Comment by luke-stdev001** [comment]
> > i'm just wondering if while we are doing this would the OCA be interested in any PR's for functional docs for using the barcode gun.
> 
> There is at OCA a new "functional" group composed essentially by functional people. Their objective is to enhance OCA procedures, documentation to attract more people using and contributing to OCA. Maybe that could fit their needs.

ok, perfect, thank you for your help. How can I get involved there, do I need to sign up for a mailing list to join discussions?

**Comment by rousseldenis** [comment]
> > > i'm just wondering if while we are doing this would the OCA be interested in any PR's for functional docs for using the barcode gun.
> > 
> > 
> > There is at OCA a new "functional" group composed essentially by functional people. Their objective is to enhance OCA procedures, documentation to attract more people using and contributing to OCA. Maybe that could fit their needs.
> 
> ok, perfect, thank you for your help. How can I get involved there, do I need to sign up for a mailing list to join discussions?

You can find information there : https://odoo-community.org/oca-for-functional-consultants

**Comment by giarve** [comment]
> > We also plan to add a few new features such as Inventory adjustments based on stock_cycle_count, something like shopfloor_stock_cycle_count, would this be something the OCA would be interested in receiving a PR for?
> 
> Of course. We had not migrated I think the normal inventory part to v16 and that should be great to begin by that part (as I suppose cycle_count is an exension of it)

Which module is it for the inventory scenarios? Is it in previous branches?

EDIT: found it! https://github.com/OCA/wms/pull/428

**Comment by luke-stdev001** [comment]
@giarve ,

It looks like this was refactored here:
[https://github.com/akretion/wms/tree/14-inventory-scenario-rebased](https://github.com/akretion/wms/tree/14-inventory-scenario-rebased)

<img width="1016" alt="Screenshot 2024-10-29 at 1 12 36 pm" src="https://github.com/user-attachments/assets/1cf8cfdb-2b1a-474e-bf51-c0789b7a346b">

@bguillot when you have a spare minute, would you mind confirming whether a PR will be made to OCA/wms 14.0 with these changes? We would love to migrate these to 16.0 for our own use if the refactoring is roughly complete. Would it be safe to base our migration work on your work in that branch above? We'll then likely extend this slightly to play nicely with stock_cycle_count.
